src/filter/colordistance/colordistance.c

```diff
@@ -125,16 +125,18 @@ void f0r_update(f0r_instance_t instance, double time,
 	float b1 = inst->color.b * 255.0;
 	float r2, g2, b2;
 	int l;
+	/* Scale factor to normalize distance to 0-255 range:
+	   0.705724361914764 ≈ 255.0 / sqrt(3 * 255^2) = 255.0 / (255.0 * sqrt(3)) */
+	const float SCALE_FACTOR = 0.705724361914764;
 	while (len--) {
 		r2 = *src++;
 		g2 = *src++;
 		b2 = *src++;
-		l = (int)rint( sqrtf( powf( r1 - r2, 2 ) + powf( g1 - g2, 2 ) + powf( b1 - b2, 2 ) ) * 0.705724361914764  );
-		/* Hint 0.35320727852735 == 255.0 / sqrt( (255)**2 + (255)**2 + (255)*2 )*/
-		if ( r1 < 0 || r1 > 255 ||  g1 < 0 || g1 > 255 ||  b1 < 0 || b1	> 255 || r2 < 0 || r2 > 255 ||  g2 < 0 || g2 > 255 || b2 < 0 || b2 > 255 ) {
-			printf ("%f %f %f\n", r2, g2, b2 );
-		}
+		l = (int)rint( sqrtf( powf( r1 - r2, 2 ) + powf( g1 - g2, 2 ) + powf( b1 - b2, 2 ) ) * SCALE_FACTOR );
 
+		// Clamp result to valid range
+		if (l < 0) l = 0;
+		if (l > 255) l = 255;
```

Comment:
Tip (not a change requested yet), but "frei0r/math.h" defines a CLAMP macro.

```suggestion
		l = CLAMP(l, 0, 255);
```